nimbus-core/src/main/java/com/antheminc/oss/nimbus/domain/model/state/repo/AbstractWSModelRepository.java

```diff
@@ -0,0 +1,128 @@
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+
+import com.antheminc.oss.nimbus.FrameworkRuntimeException;
+import com.antheminc.oss.nimbus.context.BeanResolverStrategy;
+import com.antheminc.oss.nimbus.domain.model.state.repo.db.SearchCriteria;
+import com.antheminc.oss.nimbus.support.JustLogit;
+
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter @Setter
+public abstract class AbstractWSModelRepository {
+	
+	private final BeanResolverStrategy beanResolver;
+	private final RestTemplate restTemplate;
+	private final static JustLogit logit = new JustLogit(AbstractWSModelRepository.class);
+	public AbstractWSModelRepository(BeanResolverStrategy resolver) {
+		this.beanResolver = resolver;
+		this.restTemplate = beanResolver.get(RestTemplate.class);
+	}
+
+	abstract public String getTargetUrl(String alias);
+	
+	abstract public String constructSearchUri(SearchCriteria<?> searchCriteria);
+	
+	abstract public <ID extends Serializable, T> T handleGet(ID id, Class<T> referredClass, URI url);
+	
+	abstract public <T> Object handleSearch(Class<T> referredDomainClass, Supplier<SearchCriteria<?>> criteriaSupplier, String alias, URI uri);
+	
+	protected URI createUriForAlias(String alias, String url) {
+		String targetUrl = getTargetUrl(alias);
+		if(targetUrl == null)
+			return null;
+		
+		String urlToConstruct = StringUtils.startsWith(url, "/") ? url : "/".concat(url);
+		urlToConstruct = targetUrl.concat(urlToConstruct);
+		try {
+			URI uri = new URI(urlToConstruct);
+			return uri;
+		} catch (URISyntaxException e) {
+			throw new FrameworkRuntimeException("Cannot create URI from supplied url: "+url);
+		}
+	}
+	
+	public <ID extends Serializable, T> T _get(ID id, Class<T> referredClass, String alias, String url) {
+		URI uri = createUriForAlias(alias, url);
+		if(uri == null)
+			return null;
+
+		try {
+			return handleGet(id, referredClass, uri);
+		} catch(Exception e) {
+			logit.error(() -> "Exception occured while calling Web service "+e.getMessage());
+			throw new FrameworkRuntimeException(e.getMessage(), e);
+		}
+	}
+	
+	public <T> Object _search(Class<T> referredDomainClass, String alias, Supplier<SearchCriteria<?>> criteriaSupplier) {
+		SearchCriteria<?> searchCriteria = criteriaSupplier.get();
+		String searchUri = constructSearchUri(searchCriteria);
+		URI uri = createUriForAlias(alias, searchUri);
+		if(uri == null)
+			return null;
+		
+		return handleSearch(referredDomainClass, criteriaSupplier, alias, uri);
+	}
+	
+	protected Object execute(Supplier<RequestEntity<?>> reqEntitySupplier, Supplier<ParameterizedTypeReference<?>> responseTypeSupplier) {
+		 try {
+			ResponseEntity<?> responseEntity = getRestTemplate().exchange(reqEntitySupplier.get(), responseTypeSupplier.get());
+			return Optional.of(responseEntity).map((response)->response.getBody()).orElse(null);
+		} catch(Exception e) {
+			handleException(e, reqEntitySupplier.get().getUrl());
```

Comment:
Let the exception handling method return the exception and this line can explicitly throw that exception. Otherwise it gives an impression that you are handling the exception, but in reality you are just throwing the exception back. Would recommend renaming the method to what its doing like "throw throwEx(...)"